client/state/themes/selectors.js

```diff
@@ -45,6 +45,15 @@ export const getTheme = createSelector(
 		if ( siteId === 'wpcom' || siteId === 'wporg' ) {
 			return theme;
 		}
+
+		if ( ! theme ) {
```

Comment:
Maybe we could also add this new logic to the tests [here](https://github.com/Automattic/wp-calypso/blob/cbd17f820d3192cb869a52896c3de56d28daae1c/client/state/themes/test/selectors.js#L104)